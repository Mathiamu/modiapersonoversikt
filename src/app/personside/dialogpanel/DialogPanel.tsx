import * as React from 'react';
import styled from 'styled-components';
import { Undertittel } from 'nav-frontend-typografi';
import ErrorBoundary from '../../../components/ErrorBoundary';
import { useAppState, useRestResource } from '../../../utils/customHooks';
import SendNyMeldingContainer from './sendMelding/SendNyMeldingContainer';
import FortsettDialogContainer from './fortsettDialog/FortsettDialogContainer';
import useTildelteOppgaver from '../../../utils/hooks/useTildelteOppgaver';
import { hasData } from '../../../rest/utils/restResource';
import { useDispatch } from 'react-redux';
import { setValgtTraadDialogpanel } from '../../../redux/oppgave/actions';
import { loggError } from '../../../utils/frontendLogger';
import { RouteComponentProps, withRouter } from 'react-router';
import { useInfotabsDyplenker } from '../infotabs/dyplenker';
import { useEffect } from 'react';

const DialogPanelWrapper = styled.article`
    flex-grow: 1;
`;

function DialogPanel(props: RouteComponentProps) {
    const dialogpanelTraad = useAppState(state => state.oppgaver.dialogpanelTraad);
    const tildelteOppgaver = useTildelteOppgaver();

    const slåOppOppgave = useVisTraadTilknyttetPlukketOppgave(props, dialogpanelTraad);
    if (slåOppOppgave.pending) {
        return slåOppOppgave.placeholder;
    }

    const tilknyttetOppgave = dialogpanelTraad
        ? tildelteOppgaver.paaBruker.find(oppgave => oppgave.henvendelseid === dialogpanelTraad.traadId)
        : undefined;

    return (
        <ErrorBoundary boundaryName="Dialogpanel">
            <DialogPanelWrapper role="region" aria-label="Dialogpanel">
                <Undertittel className="sr-only">Dialogpanel</Undertittel>
                {dialogpanelTraad !== undefined ? (
                    <FortsettDialogContainer
                        traad={dialogpanelTraad}
                        tilknyttetOppgave={tilknyttetOppgave}
                        key={dialogpanelTraad.traadId}
                    />
                ) : (
                    <SendNyMeldingContainer />
                )}
            </DialogPanelWrapper>
        </ErrorBoundary>
    );
}

export default withRouter(DialogPanel);
