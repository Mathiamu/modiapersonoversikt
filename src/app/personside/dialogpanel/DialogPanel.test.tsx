import * as React from 'react';
import * as renderer from 'react-test-renderer';
import TestProvider from '../../../test/Testprovider';
import DialogPanel from './DialogPanel';

beforeEach(() => {
    Date.prototype.getTime = jest.fn(() => 0);
    Date.parse = jest.fn(() => 0);
});

test('viser dialogpanel', () => {
    const dialogPanelBody = renderer.create(
        <TestProvider>
            <DialogPanel />
        </TestProvider>
    );

    expect(dialogPanelBody.toJSON()).toMatchSnapshot();
});
