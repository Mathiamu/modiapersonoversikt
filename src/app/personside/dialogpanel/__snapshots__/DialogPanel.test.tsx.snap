// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`viser dialogpanel 1`] = `
.c1 {
  -webkit-filter: drop-shadow(0 0 1px rgba(0,0,0,0.7));
  filter: drop-shadow(0 0 1px rgba(0,0,0,0.7));
}

.c1 .ekspanderbartPanel__innhold {
  padding: 0;
}

.c1 .ekspanderbartPanel__hode:hover {
  box-shadow: inset 0 0 0 0.1rem #0067c5,0 0.1rem 0.1rem rgba(0,0,0,0.2);
}

.c1 label {
  position: absolute !important;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-clip: rect(1px 1px 1px 1px);
  clip: rect(1px 1px 1px 1px);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
}

.c0 {
  border-top: rgba(0,0,0,0.1) 1px solid;
  border-bottom: rgba(0,0,0,0.1) 1px solid;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c0 > *:not(:last-child) {
  margin-bottom: .8rem;
}

<div
  aria-label="Dialogpanel"
  className="c0"
  role="region"
>
  <div
    className="c1"
  >
    <div
      className="ekspanderbartPanel ekspanderbartPanel--lukket"
    >
      <button
        aria-expanded={false}
        className="ekspanderbartPanel__hode"
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        <div
          className="ekspanderbartPanel__flex-wrapper"
        >
          <span
            className="typo-undertittel ekspanderbartPanel__heading"
          >
            Hurtigreferat
          </span>
          <span
            className="ekspanderbartPanel__indikator"
          />
        </div>
      </button>
    </div>
  </div>
  <h2
    className="typo-undertittel"
  >
    Dialogpanel
  </h2>
  <p
    className="typo-normal"
  >
    Lorem ipsum dolor sit amet, consectetur adipisicing elit. Minus, veniam?
  </p>
</div>
`;
